src/server/endpoints/friendHandler.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+const HTTP_STATUSES = require('../modules/httpStatuses');
+const friendsCollection = require('../collections/friendsDatabase');
+const usersCollection = require('../collections/usersDatabase');
+const {getAccessToken} = require('../modules/tokenHandler');
+
+function findFriends(idArray, res) {
+  usersCollection.findUsers(idArray, (err, result) => {
+    if (err !== null) {
+      res.status(HTTP_STATUSES.SERVER_ERROR).json({errorType: 'serverError'});
+      return;
+    }
+    let friendsInfo = [];
+
+    result.forEach(function(user) {
+      let newUser = {};
+
+      newUser._id = user._id;
+      newUser.username = user.username;
+      newUser.userPicURL = user.userPicURL;
+      newUser.numberOfFriends = user.numberOfFriends;
+      friendsInfo = friendsInfo.concat(newUser);
+    });
+    let obj = {friends: friendsInfo};
```

Comment:
you dont need to create a new variable for this, you can pass the map return value to the json function. Actually you declared 3 variables, however you can do  it without any variables